app/scripts/directives/podDonut.js

```diff
@@ -153,7 +154,16 @@ angular.module('openshiftConsole')
             return 'Terminating';
           }
 
+          var warnings = podWarningsFilter(pod);
           if (isTroubledPodFilter(pod)) {
```

Comment:
Shouldn't need to call `isTroubledPodFilter` any longer if you have the warnings. If warnings is not empty, it's troubled.